numpy/random/bounded_integers.pyx.in

```diff
@@ -50,13 +50,21 @@ cdef object _rand_{{nptype}}_broadcast(np.ndarray low, np.ndarray high, object s
     cdef int buf_rem = 0
 
     # Array path
+    is_open = not closed
     low_arr = <np.ndarray>low
     high_arr = <np.ndarray>high
     if np.any(np.less(low_arr, {{lb}})):
         raise ValueError('low is out of bounds for {{nptype}}')
-    if np.any(np.greater(high_arr, {{ub}})):
+    if closed:
+        high_comp = np.greater_equal
+        low_high_comp = np.greater
+    else:
+        high_comp = np.greater
+        low_high_comp = np.greater_equal
+
+    if np.any(high_comp(high_arr, {{ub}})):
         raise ValueError('high is out of bounds for {{nptype}}')
-    if np.any(np.greater_equal(low_arr, high_arr)):
+    if np.any(low_high_comp(low_arr, high_arr)):
         raise ValueError('low >= high')
```

Comment:
This error message doesn't match the comparator

Reply:
I think that string probably needs to be dynamic and change depending on whether closed or not.

Reply:
low=high is ok for closed, but only low<high for open.

Reply:
This has been fixed.